libkbfs/md_journal.go

```diff
@@ -196,13 +196,15 @@ func makeMDJournalWithIDJournal(
 	}
 
 	if earliest != nil {
-		if earliest.BID() != latest.BID() {
+		if earliest.BID() != latest.BID() &&
+			!(earliest.BID() == NullBranchID &&
```

Comment:
The prefix of the journal containing previous local squashes do not get converted to a new local squash branch; they remain with the old `NullBranchID` bid.